cloudfoundry-operations/src/main/java/org/cloudfoundry/operations/applications/DefaultApplications.java

```diff
@@ -431,6 +461,142 @@ public Mono<Void> pushManifest(PushApplicationManifestRequest request) {
             .checkpoint();
     }
 
+    @Override
+    public Mono<Void> pushManifestV3(PushManifestV3Request request) {
+        byte[] manifestSerialized;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            ApplicationManifestUtilsV3.write(baos, request.getManifest());
+            manifestSerialized = baos.toByteArray();
+        } catch (IOException e) {
+            throw new RuntimeException("Could not serialize manifest", e);
+        }
+
+        return Mono.zip(this.cloudFoundryClient, this.spaceId)
+            .flatMap(function((cloudFoundryClient, spaceId) ->
+                applyManifestAndWaitForCompletion(cloudFoundryClient, spaceId, manifestSerialized)
+                    .then(Mono.just(Tuples.of(cloudFoundryClient, spaceId)))))
+            .flatMapMany(function((cloudFoundryClient, spaceId) -> Flux.fromIterable(request.getManifest().getApplications())
+                .map(manifestApp -> Tuples.of(cloudFoundryClient, spaceId, manifestApp))))
+            .flatMap(function((cloudFoundryClient, spaceId, manifestApp) ->
+                getApplicationIdV3(cloudFoundryClient, manifestApp.getName(), spaceId)
+                    .flatMap(appId -> Mono.zip(Mono.just(appId), createPackage(cloudFoundryClient, appId, manifestApp)))
+                    .flatMap(function((appId, packageId) ->
+                        buildAndStage(cloudFoundryClient, manifestApp, packageId)
+                            .flatMap(dropletId -> applyDropletAndWaitForRunning(cloudFoundryClient, manifestApp.getName(), appId, dropletId))))))
+            .then();
+    }
+
+    private static Mono<Void> applyDropletAndWaitForRunning(CloudFoundryClient cloudFoundryClient, String appname, String appId, String dropletId) {
```

Comment:
Move private methods to bottom of class and make static (same style as the other privet methods)